### PR TITLE
CSPL-2165 - Static analysis bug fixes for staticcheck against operator code base

### PR DESCRIPTION
**Pull request by akondur**
First stab at fixing static analysis bugs generated by `staticcheck` against operator code base.

### CODE REVIEW

**Comment by akondur** [comment]
Static analysis rerports
[golangci-lint_report.log](https://github.com/splunk/splunk-operator/files/10072635/golangci-lint_report.log)
[staticcheck_report.log](https://github.com/splunk/splunk-operator/files/10072636/staticcheck_report.log)

**Review by vivekr-splunk — APPROVED** [review]
LGTM

File: .github/workflows/prodsec-workflow.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@ on:
     branches:
     - main
     - develop
+    - cspl_2165_fix_sa_bugs
```
Comment (jryb) [inline .github/workflows/prodsec-workflow.yml:7]: Probably dont need this.

File: controllers/suite_test.go [hunk 2]
```diff
@@ -172,89 +166,3 @@ var _ = AfterSuite(func() {
 	By("tearing down the test environment")
 	testEnv.Stop()
 })
-
-func mainFunction(scheme *runtime.Scheme) (manager.Manager, error) {
```
Comment (jryb) [inline controllers/suite_test.go:176]: No need for main anymore?

File: pkg/splunk/enterprise/util.go [hunk 3]
```diff
@@ -234,16 +233,6 @@ func ApplySplunkConfig(ctx context.Context, client splcommon.ControllerClient, c
 	return namespaceScopedSecret, nil
 }
 
-// getIndexerExtraEnv returns extra environment variables used by indexer clusters
```
Comment (jryb) [inline pkg/splunk/enterprise/util.go:237]: I take it these are removed due to not being used.  So this functionality has moved or is not longer needed?

**Review by jryb — APPROVED** [review]

File: .github/workflows/prodsec-workflow.yml (hunk 1 above)
Comment (akondur) [reply .github/workflows/prodsec-workflow.yml:7]: Yes will be removing this after review approvals.

File: controllers/suite_test.go (hunk 2 above)
Comment (akondur) [reply controllers/suite_test.go:176]: It is currently being unused. @vivekr-splunk Do we need to keep it?

File: pkg/splunk/enterprise/util.go (hunk 3 above)
Comment (akondur) [reply pkg/splunk/enterprise/util.go:237]: Currently we don't need the function/functionality. Per the [PR](https://github.com/splunk/splunk-operator/pull/244), we can reintroduce it if required.

File: controllers/suite_test.go (hunk 2 above)
Comment (akondur) [reply controllers/suite_test.go:176]: Spoke with @vivekr-splunk , we can remove for now and revisit later if required.

**Pull request merged**